python/ray/train/tests/test_base_trainer.py

```diff
@@ -204,38 +204,59 @@ def __init__(self):
         def warning(self, msg):
             self.warnings.append(msg)
 
+        def warn(self, msg, **kwargs):
+            self.warnings.append(msg)
+
         def info(self, msg):
             print(msg)
 
     try:
-        old = base_trainer.logger
-        base_trainer.logger = MockLogger()
+        old = tuner_internal.warnings
+        tuner_internal.warnings = MockLogger()
```

Comment:
the reason why we want to add introspection at the Tune level is because then we can take into account num samples and concurrency to provide a better warning.